orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/tasks/CreateWebhookTaskSpec.groovy

```diff
@@ -280,6 +280,39 @@ class CreateWebhookTaskSpec extends Specification {
     stage.context.statusEndpoint == null
   }
 
+  // Tests https://github.com/spinnaker/spinnaker/issues/2163
+  def "should evaluate statusUrlJsonPath for differing payloads"() {
+    setup:
+    def stage = new Stage(pipeline, "webhook", "My webhook", [
+      url: "https://my-service.io/api/",
+      waitForCompletion: "true",
+      statusUrlResolution: "webhookResponse",
+      statusUrlJsonPath: 'concat("https://my-service.io/api/id/", $.id)'
+    ])
+
+    when:
+    createWebhookTask.webhookService = Stub(WebhookService) {
+      exchange(HttpMethod.POST, "https://my-service.io/api/", null, null) >> new ResponseEntity<Map>([ success: true, id: "1" ], HttpStatus.CREATED)
+    }
+    def result1 = createWebhookTask.execute(stage)
+
+    then:
+    result1.status == ExecutionStatus.SUCCEEDED
+    result1.context.webhook.statusEndpoint == "https://my-service.io/api/id/1"
+
+    setup:
```

Comment:
this empty `setup:` could be removed

Reply:
Played with it a bit locally ... the test might be more simple written as 

```
given:
    def stage = new Stage(pipeline, "webhook", "My webhook", [
      url: "https://my-service.io/api/",
      waitForCompletion: "true",
      statusUrlResolution: "webhookResponse",
      statusUrlJsonPath: 'concat("https://my-service.io/api/id/", $.id)'
    ])

    createWebhookTask.webhookService = Stub(WebhookService) {
      exchange(HttpMethod.POST, "https://my-service.io/api/", null, null) >>> [
        new ResponseEntity<Map>([ success: true, id: "1" ], HttpStatus.CREATED),
        new ResponseEntity<Map>([ success: true, id: "2" ], HttpStatus.CREATED)
      ]
    }

    when:
    def result1 = createWebhookTask.execute(stage)
    def result2 = createWebhookTask.execute(stage)

    then:
    result1.status == ExecutionStatus.SUCCEEDED
    result1.context.webhook.statusEndpoint == "https://my-service.io/api/id/1"

    result2.status == ExecutionStatus.SUCCEEDED
    result2.context.webhook.statusEndpoint == "https://my-service.io/api/id/2"
```

Had to lookup how to have multiple invocations from a `Stub` ... the `>>>` lets you do that.

Reply:
`>>>` oooh fancy